src/app/modules/form/__tests__/form.service.spec.ts

```diff
@@ -111,4 +111,102 @@ describe('FormService', () => {
       expect(actualResult._unsafeUnwrapErr()).toBeInstanceOf(DatabaseError)
     })
   })
+
+  describe('retrieveFormById', () => {
+    it('should return form successfully', async () => {
+      // Arrange
+      const formId = new ObjectId().toHexString()
+      const expectedForm = {
+        _id: formId,
+        title: 'mock title',
+        admin: new ObjectId(),
+      } as IFormSchema
+      const retrieveFormSpy = jest
+        .spyOn(Form, 'findById')
+        .mockImplementationOnce(
+          () =>
+            (({
+              exec: jest.fn().mockResolvedValue(expectedForm),
+            } as unknown) as mongoose.Query<any>),
+        )
```

Comment:
might be able to simplify this using `mockReturnValueOnce`

Reply:
wonderful, thanks!